File: Sorting-with-Twos.cpp
// Author : ysh
// 2023/12/05 Tue 20:33:26
// https://codeforces.com/problemset/problem/1891/A
#include<bits/stdc++.h>
using namespace std;
int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int n;cin>>n;
    while(n--) {
        int n;cin>>n;
        vector<int>f(n);
        for(int &i : f) cin>>i;

        int l = 0,r = 1;
        while(r <= n) {
            if(!is_sorted(f.begin() + l,f.begin() + r)) goto no;
            l = l << 1;
            r = r << 1;
            if(l == 0) l = 1;
        }
        if(!is_sorted(f.begin() + l,f.end())) goto no;

        yes:
        cout<<"YES\n";
        continue;

        no:
        cout<<"NO\n";
    }
}

File: Sorting-with-Twos.py
# Author : ysh
# 2023/12/05 Tue 20:41:19
# https://codeforces.com/problemset/problem/1891/A
for i in range(int(input())):
    n = int(input())
    f = [int(i) for i in input().split()]
    l = 0;r = 1;
    c = True
    while r <= n:
        for i in range(l + 1,r):
            if f[i] < f[i - 1]:
                c = False
        l = l << 1
        r = r << 1
        if l == 0: l = 1;
    for i in range(l + 1,n):
        if f[i] < f[i - 1]:
            c = False
    if c: print('YES')
    else: print('NO')
quit()